src/main/java/brave/opentracing/BraveSpan.java

```diff
@@ -94,6 +94,17 @@ public Span setTag(String key, String value) {
             delegate.kind(brave.Span.Kind.CLIENT);
         } else if (Tags.SPAN_KIND.getKey().equals(key) && Tags.SPAN_KIND_SERVER.equals(value)) {
             delegate.kind(brave.Span.Kind.SERVER);
+        } else if (Tags.SAMPLING_PRIORITY.getKey().equals(key)) {
```

Comment:
I see your concern about `abandon` however this would ensure that "some" descendants would by sampled.. but it is ambiguous.